crates/revm/src/optimism/processor.rs

```diff
@@ -70,6 +69,9 @@ impl<'a> BlockExecutor for EVMProcessor<'a> {
         let is_regolith =
             self.chain_spec.fork(Hardfork::Regolith).active_at_timestamp(block.timestamp);
 
+        super::ensure_create2_deployer(self.chain_spec().clone(), block.timestamp, self.db_mut())
+            .map_err(|_| BlockExecutionError::ProviderError)?;
```

Comment:
if the fork block ends up being empty, than the create2 deployer will never be committed because of an early return few lines above

Reply:
Thankfully this should never be possible :)

Optimism blocks will always have at least one transaction, which is the L1 info transaction sent by the system.  Added a clarifying comment and removed that check since we'll never need it.